src/components/swap/SwapBox.vue

```diff
@@ -245,17 +245,158 @@ export default defineComponent({
         });
         console.info("approval receipt", receipt);
 
-        // set allowance to true
         this.hasAllowance = true;
         this.showSpinner = false;
       } catch (err) {
-        // set allowance to false
         this.showSpinner = false;
         this.hasAllowance = false;
 
         console.error("approval error: ", err);
       }
     },
+    async onSwap() {
+      if (!this.swapFrom.value) {
+        console.error("You need to estimate a swap amount!");
+        return;
+      }
+
+      this.showSpinner = true;
+
+      const estimatedGasFee = await fetch(
+        `${process.env.VUE_APP_RELAYER_ENDPOINT}/estimated-gas`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            amount: this.swapFrom.value,
+            unitType: "wei",
+          }),
+        }
+      );
+
+      const signedData = await this.signWithMetamask(estimatedGasFee); // needs to match what is in contract
+      // calll swap
+      if (!signedData) {
+        // todo: display error
+        console.error("no signed data!");
+        return null;
+      }
+
+      try {
+        const response = await fetch(
+          // 0xCa1D8715989D70173DdaE7e7b2111f49dE88d1c4
+          `${process.env.VUE_APP_RELAYER_ENDPOINT}/swap`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              swapData: signedData.message.swapData,
+              deadline: signedData.message.deadline,
+              relayerAddress: process.env.VUE_APP_RELAYER_ADDRESS,
+              v: signedData.v,
+              r: signedData.r,
+              s: signedData.s,
+              estimatedGasFee,
+              sideTokenBtcContractAddress: this.swapFrom.address,
+            }),
+          }
+        );
+      } catch (err) {
+        console.error("couldnt swap", err);
+      } finally {
+        this.showSpinner = false;
+      }
+    },
+    async signWithMetamask(estimatedGas) {
+      const contract = new this.web3.eth.Contract(
+        BRIDGE_ABI,
```

Comment:
Remove BRIDGE, you can take a look on this script to do the swap.
https://github.com/rsksmart/tokenbridge-backend/blob/main/api/scripts/scriptSwapRwbtc.ts

Reply:
removed since we dont use it at all, thanks!